src/components/DropdownProblem.tsx

```diff
@@ -1,7 +1,91 @@
 import { BaseProblemProps } from './ProblemSetBlock'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import React, { useCallback, useState } from 'react'
+import * as Yup from 'yup'
 
-interface DropdownProblemProps extends BaseProblemProps {}
+interface DropdownProblemProps extends BaseProblemProps {
+  solutionOptions: string
+}
+
+interface DropdownFormValues {
+  response: string
+}
+
+export const DropdownProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback, content, buttonText, solutionOptions,
+  encourageResponse, correctResponse, solution, retryLimit
+}: DropdownProblemProps): JSX.Element => {
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+
+  const schema = Yup.object({
+    response: Yup.string().trim().required('Please select an answer')
+  })
+
+  const generateOptions = (): JSX.Element[] => {
+    const options: JSX.Element[] = []
+    const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+    options.push(<option key="initial" value="">Select an answer</option>)
+    parsedOptionValues.forEach(val => options.push(<option key={val} value={val}>{val}</option>))
+
+    return options
+  }
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const handleSubmit = async (values: DropdownFormValues): Promise<void> => {
+    if (values.response === solution) {
+      setFeedback(correctResponse)
+      solvedCallback()
+      setFormDisabled(true)
+    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
+      setRetriesAllowed(currRetries => currRetries + 1)
+      setFeedback(encourageResponse)
+      allowedRetryCallback()
+    } else {
+      exhaustedCallback()
+      setFeedback('No more attempts allowed')
+      setFormDisabled(true)
+    }
+  }
 
-export const DropdownProblem = ({ solvedCallback, exhaustedCallback, allowedRetryCallback }: DropdownProblemProps): JSX.Element => {
-  return <></>
+  return (
+    <div className="os-raise-bootstrap">
+      <div className="my-3" ref={contentRefCallback} dangerouslySetInnerHTML={{ __html: content }} />
+      <Formik
+        initialValues={{ response: '' }}
+        onSubmit={handleSubmit}
+        validationSchema={schema}
+      >
+        {({ isSubmitting, values, setFieldValue }) => (
+          <Form>
+            <Field
+              name="response"
+              as="select"
+              value={values.response}
+              disabled={isSubmitting || formDisabled}
+              className="os-form-select mb-3"
+              onChange={(e: React.ChangeEvent<HTMLSelectElement>) => { clearFeedback(); setFieldValue('response', e.target.value) }}
```

Comment:
So keeping in mind that the `<Field>` component here is really just a `formik` wrapper around the underlying `<select>` element, this is basically just passing the attribute through to that. So your question then maps to when is `onChange` invoked on a `<select>` element. You can find the corresponding documentation on that [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event), but basically this is just when the dropdown value is changed. Typically `formik` will glue all of these onChange / onBlur / ... handlers for you and do things like set the internal state values, do validation, etc. But here I wanted to add some custom business logic of clearing the feedback so I'm doing that and then manually invoking `setFieldValue()` per note on custom change handlers [here](https://formik.org/docs/api/formik#setfieldvalue-field-string-value-any-shouldvalidate-boolean--void).

Reply:
Are you clearing feedback so that when a response is being shown and then the user re-selects something else, they are not still confused by the feedback response? If so, I get that, that's a nice touch. Otherwise the feedback will remain until the next submission, right?

Reply:
@rnathuji I'm now thinking that I should have a custom change handler as well for MultiselectProblems, since my formik value model is structured with just one array of strings `{response: string[]}`. My idea here is to create a function `updateResponse()` that takes in the formik model and the new value, and pops/pushes strings to `formik.values.response` accordingly. Does that make sense? Or do am I complicating things?